### DISCUSSION TITLE
Models with additional fields that are not defined in the schema

### DISCUSSION THREAD

**Discussion by eglanddd** [sensedeep/dynamodb-onetable#273] [Q&A] [answered]
How to get all item fields including those that are not defined in the schema and how to create items with additional fields that are not defined in the schema?

**Comment by mobsense** [comment]
You can use the Table.*Items APIs:

```
    deleteItem(properties: OneProperties, params?: OneParams): Promise<void>;
    getItem(properties: OneProperties, params?: OneParams): Promise<AnyEntity | undefined>;
    putItem(properties: OneProperties, params?: OneParams): Promise<AnyEntity>;
    queryItems(properties: OneProperties, params?: OneParams): Promise<Paged<AnyEntity>>;
    scanItems(properties?: OneProperties, params?: OneParams): Promise<Paged<AnyEntity>>;
    updateItem(properties: OneProperties, params?: OneParams): Promise<AnyEntity>;
```

Usage:
```
let items = await table.queryItems({pk: 'account:AcmeCorp'}, {parse: true, hidden: true})
```

See the readme for doc.

**Reply by eglanddd** [reply]
Can this be done from Models APIs?

**Reply by mobsense** [reply]
Why do you want to use the model APIs?  They enforce the schema for the model.

**Reply by eglanddd** [reply]
When using the models APIs the PK is formed by just giving the fields which the PK is formed. It would be nice if we could get the raw item and create an item with additional fields that are not defined in the schema from the models APIs.

**Comment by mobsense** [comment, marked as answer]
You can use params.parse == false to get the raw item.
